daos/GroupDao.ts

```diff
@@ -0,0 +1,94 @@
+/**
+ * @file Implements DAO managing data storage of groups. Uses mongoose GroupModel
+ * to integrate with MongoDB
+ */
+import GroupDaoI from "../interfaces/group/GroupDaoI";
+import Group from "../models/group/Group";
+import GroupModel from "../mongoose/groups/GroupModel"
+
+/**
+ * @class GroupDao Implements Data Access Object managing data storage
+ * of Groups
+ * @property {GroupDao} groupDao Private single instance of GroupDao
+ */
+export default class GroupDao implements GroupDaoI {
+    private static groupDao: GroupDao | null = null
+
+    /**
+     * Creates singleton DAO instance
+     * @returns {GroupDao} GroupDao
+     */
+    public static getInstance = (): GroupDao => {
+        if (GroupDao.groupDao === null) {
+            GroupDao.groupDao = new GroupDao()
+        }
+        return GroupDao.groupDao
+    }
+
+    private constructor() {
+    }
+
+    /**
+     * Removes group object with given primary key from database.
+     * @param {string} uid Primary key of user
+     * @param {string} gid Primary key of group
+     * @returns Promise To be notified when group is removed from the database
+     */
+    deleteGroup = async (uid: string, gid: string): Promise<any> =>
+        GroupModel.deleteOne({_id: gid})
+
+    /**
+     * Finds all groups with both active user and other user
+     * @param {string} active_uid Primary key of active user
+     * @param {string} other_uid Primary key of other user
+     * @returns Promise To be notified when group is retrieved from database
+     */
+    findAllCommonGroupsAnotherUser = async (active_uid: string, other_uid: string): Promise<Group[]> =>
```

Comment:
Same with here - do we need to pass active uid or can we just get from the session info?

Reply:
We will need to check Group object if both user in session and other id is in group so i think we might need it

Reply:
I meant just for the active user id cause I think we can get that from the session info but agree we'll definitely still need the other id.
Made a post below about testing difficulties without passing both though so let me know what you think

Reply:
I tested GroupModel.find({members: {$all: [active_uid, other_uid]}}) in mongo shell with some mock groups and was able to find all groups with both users in it. I think just having the other_uid will give us all groups other_uid is in and not the current user. We could use the session info to pass in the current user's id and return mutual groups